Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php

```diff
@@ -61,7 +61,7 @@ public function generateFilter($name, $value, $documentName)
         if (2 == count($columnsTree)) {
             list($property, $referenceProperty) = $columnsTree;
 
-            $metadata = $this->documentManager->getClassMetadata($documentName);
+            $metadata = $this->documentManager->getMetadataFactory()->getMetadataFor($documentName);
```

Comment:
Why would you suppress the internal call ?

This way you must be aware of the metadatFactory (which should not be the case )